invenio_users_resources/services/groups/service.py

```diff
@@ -25,7 +28,8 @@ def read(self, identity, id_):
         # resolve and require permission
         group = GroupAggregate.get_record_by_name(id_)
         if group is None:
-            raise LookupError(f"No group with id '{id_}'.")
+            # return 403 even on empty resource due to security implications
+            raise GroupNotFoundError(_("Permission denied."))
```

Comment:
I would just use the [PermissionDenied](https://github.com/inveniosoftware/invenio-records-resources/blob/c79022501269bf0c01eb026d3a1f8bf7f059d25c/invenio_records_resources/services/errors.py#L16) class then and If I am not mistaken this is already handled [here](https://github.com/inveniosoftware/invenio-records-resources/blob/c79022501269bf0c01eb026d3a1f8bf7f059d25c/invenio_records_resources/resources/errors.py#L116)